### DISCUSSION TITLE
Ordered task lists

### DISCUSSION THREAD

**Discussion by dimrozakis** [squidfunk/mkdocs-material#3588] [Ideas]
Hi @squidfunk 

I see that the [documentation regarding task lists](https://squidfunk.github.io/mkdocs-material/reference/lists/#using-task-lists), only mentions unordered task lists. I tried defining an ordered task list and it works, but is displayed the same as an unordered one.

I work mostly on Gitlab which renders ordered task lists with the number prefixes shown. I was just now a bit surprised to see that Github doesn't display the number, just like mkdocs-material.

I think I'd like to be able to use ordered task lists (and have them displayed as ordered) as I do in Gitlab. Is the display of ordered task lists the same as unordered task lists a deliberate design choice for mkdocs-material? Shouldn't perhaps the markdown author's choice to use ordered task lists be respected when displaying that list, similar to using a regular ordered vs unordered list?

**Comment by squidfunk** [comment]
Thanks for suggesting. How should ordered task lists work? How should they look? Looping in @facelessuser.

**Reply by facelessuser** [reply]
If pymdown-extensions is rendering ordered tasklists as normal tasklists, it sounds like a Material style issue. If they aren't rendering at all, then it would be a feature request also requiring a Material style update I assume.

I'd have to double check, but it wouldn't surprise me if we allow them and are just not styling them with the numbers. When I created the extension, I used GitHub as the reference.

**Reply by dimrozakis** [reply]
Thank you both for your answers.

This is how ordered task lists appear in mkdocs-material:

![Screenshot from 2022-02-17 03-15-50](https://user-images.githubusercontent.com/5428237/154385411-04f7d295-3f71-41b3-b232-64c2b0386e5c.png)

From the source it can be seen that the first list uses `<ul>` and the second uses `<ol>` but are styled identically.

For reference, this is the same markdown rendered in Gitlab:

![Screenshot from 2022-02-17 03-14-59](https://user-images.githubusercontent.com/5428237/154385430-c0f22f76-ca78-4816-aa27-7386eb7338d7.png)

The checks might not be quite as pretty as is the case with mkdocs-material but the list items are numbered.

>How should ordered task lists work? How should they look?

They should be numbered, identically to regular ordered lists, but also contain the checkbox. I think that the number should come first, then the checkbox, like Gitlab does. That way, if only some items have checkboxes, as is the case with the example above, the numbers (which all items would have since this is an ordered list) would come first and be aligned with each other.

Please correct me if I'm wrong but this seems solvable with CSS.

If that's the case then I could certainly get this behavior with custom CSS. Leaving aside the fact I'd have to do this to several repos, I think that:

1. (as mentioned in my initial comment) The markdown author's choice to use ordered task lists should be respected when displaying them
2. It's unlikely to cause backwards compatibility issues since
    1. ordered task lists weren't even officially supported (since they weren't documented)
    2. it'd seem very odd to me if someone depended on ordered task lists being rendered as unordered (why wouldn't they just use unordered then)
3. There's added value in supporting ordered task lists. Not a huge value, admittedly, but equal to the value of supporting both unordered and ordered regular lists. For example it better conveys that the items (tasks) have to be completed in order, it makes it easier to refer to a task by its number and so on.

So thank you for considering this.

**Reply by facelessuser** [reply]
Yes, this appears to be something solvable in Material directly. It appears I never restricted ordered lists from becoming task lists, I've just never actually used order lists for task lists, so I never needed to worry about their styling 🙂.

**Reply by squidfunk** [reply]
I'm quite sure that this will not play well together with the custom checkbox integration due to positioning issues, and thus currently not considering adding support for it. If somebody manages to pull off a decent implementation, I'm happy to merge it though.

**Reply by dimrozakis** [reply]
Fair enough, thanks for considering it, to be honest it's really not that big of an issue :)
